### PR TITLE
Adding Intel and STM32 flash instructions and turning board section into more generic.

### PR DESCRIPTION
**Pull request by munoz0raul**

### CODE REVIEW

**Comment by doanac** [comment]
Docs for 5e318312405f5f74013409c05e9b7951e9f24bd8 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/971/docs/artifacts/html/index.html

**Review by angolini — APPROVED** [review]
Great improvements! LGTM

**Review by vanmaegima — APPROVED** [review]
LGTM

**Comment by munoz0raul** [comment]
@mike-scott and @ricardosalveti I just sync it with the latest master. I think it is easy to accept and merge.

**Comment by doanac** [comment]
Docs for 0c80af62f9d98f61bc29bfe5818435d27ecf989d are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1006/docs/artifacts/html/index.html

File: source/reference-manual/boards/stm32mp1.rst [hunk 1]
```diff
@@ -0,0 +1,13 @@
+STM32MP157 Discovery Kit
+========================
+
+.. include:: generic-prepare.rst
```
Comment (Tim-Anderson) [inline source/reference-manual/boards/stm32mp1.rst:4]: The generic prepare is referring to the raspberrypi which might be a bit confusing

**Review by Tim-Anderson — APPROVED** [review]
just a comment about the generic prepare otherwise LGTM

File: source/reference-manual/boards/stm32mp1.rst (hunk 1 above)
Comment (munoz0raul) [reply source/reference-manual/boards/stm32mp1.rst:4]: I changed that on the PR  https://github.com/foundriesio/docs/pull/245:

https://ci.foundries.io/projects/fio-docs/builds/1016/docs/artifacts/html/reference-manual/boards/raspberrypi.html

Generic will have <machine_name> in the pictures.

**Pull request merged**